ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -716,8 +718,7 @@ static void HandleKernelRequest(void* messageMemory, uint32_t messageSize)
                 << dec << request.messageHeader->fsidInode.inode
                 << " -> '"
                 << pathBuffer
-                << "' -> relative path '"
-                << (relativePath != nullptr ? relativePath : "[NULL]")
+                << "' -> relative path '" << relativePath
```

Comment:
Why was the `nullptr` check removed here?  I noticed we're still checking on line 775.

Reply:
relativeFromPath can be null, while relativePath can not.

Two reasons:
- relativePath as nullptr can mean we need to look up path.  We don't want to confuse ourselves with it having a double meaning.  It is always sent as "" when for paths outside the virtualization root.
- realtiveFromPath is not relevant for all messages, hence the allowance of null.  For messages it is relevant for it is sent as "".

That said, on 775 fromRelativePath should never be nullptr since we are dealing with a hardlink.  It should be "".  I'll update that line good catch!

Reply:
> relativeFromPath can be null, while relativePath can not.

`relativePath` is the value returned by `GetRelativePath`, and it looks like under certain error conditions `GetRelativePath` can return return `nullptr`.

It looks like we should have been doing a better job of handing that before (e.g. logging an error and returning).  Can you add a check for that scenario (or if it's possible `GetRelativePath` will never return `nullptr` on this code path that would be good to know as well).

Reply:
As discussed offline if we can't find a Relative path, we could end up in a bad state.

Code is being updated to cleanup in return if relativepath can't be found.